sdk/include/opentelemetry/sdk/trace/batch_span_processor.h

```diff
@@ -19,7 +19,7 @@ namespace trace
  * This is an implementation of the SpanProcessor which creates batches of finished spans and passes
  * the export-friendly span data representations to the configured SpanExporter.
  */
-class BatchSpanProcessor : public SpanProcessor
+class BatchSpanProcessor : public ForkAwareSpanProcessor
```

Comment:
Shouldn't `BatchSpanProcessor` remain a platform-agnostic class? Forking is a concept very foreign to Windows..

Reply:
Per our conversation on gitter, should I instead make two separate classes? Something like `ForkAwareBatchSpanProcessor` and `ForkAwareSimpleSpanProcessor`?

This way we can leave it to the user to decide what to use

Reply:
Also, do you prefer any specific directory structure wherein we can store all ForkAware processors in one folder, and all the ForkAware exporters in another folder?

Reply:
Also, I had one general comment regarding inheriting from the `ForkAwareSpanProcessor` class. The class primarily just exposes three fork handlers that any custom processor can implement as required. 

Even if a processor (inheriting from the `ForkAwareSpanProcessor`) is used on a non-linux OS, there would not be essentially any breaks in the code, meaning that they would still function properly.

As far as registering fork handlers is concerned, that's already handled [here](https://github.com/open-telemetry/opentelemetry-cpp/blob/master/sdk/src/common/CMakeLists.txt#L3) on a platform basis. You can also see [here](https://github.com/open-telemetry/opentelemetry-cpp/blob/master/sdk/src/common/platform/fork_windows.cc) that no fork handlers are registered on the Windows platform.